lib/services/messenger/translators/discourse.ts

```diff
@@ -0,0 +1,257 @@
+/*
+ Copyright 2016-2017 Resin.io
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+
+import * as Promise from 'bluebird';
+import * as _ from 'lodash';
+import * as request from 'request-promise';
+import { ProcBotError } from '../../../framework/procbot';
+import { ProcBotErrorCode } from '../../../framework/procbot-types';
+import {
+	DiscourseConnectionDetails, DiscourseEmitInstructions,
+	DiscourseEvent, DiscourseResponse,
+} from '../../discourse-types';
+import {
+	MessageAction, MessageEvent,
+	MessageInformation, MessageResponseData, TransmitInformation,
+} from '../../messenger-types';
+import { DataHub } from '../datahubs/datahub';
+import * as Translator from './translator';
+
+export class DiscourseTranslator implements Translator.Translator {
+	private hubs: DataHub[];
+	private connectionDetails: DiscourseConnectionDetails;
+	private eventEquivalencies: {[generic: string]: string[]} = {
```

Comment:
The 'eventEquivalencies` type seems to be the same in several files. Possibly a good candidate for an interface in a `translators.d.ts`?